builtin/ui/theme.lua

```diff
@@ -0,0 +1,39 @@
+-- Luanti
+-- SPDX-License-Identifier: LGPL-2.1-or-later
+-- Copyright (C) 2023 v-rob, Vincent Robinson <robinsonvincent89@gmail.com>
+
+local prelude_theme = ui.Style {
+	ui.Style "root" {
+		pos = {1/2},
+		anchor = {1/2},
+		span = {0},
+
+		ui.Style "@backdrop" {
+			display = "hidden",
+			clip = "both",
+		},
+		ui.Style "@backdrop$focused" {
+			display = "visible",
+		},
+	},
+	ui.Style "image" {
+		icon_scale = 0,
+	},
+	ui.Style "check, switch, radio" {
+		icon_place = "left",
+	},
+}
+
+function ui.get_prelude_theme()
+	return prelude_theme
+end
+
+local default_theme = prelude_theme
+
+function ui.get_default_theme()
+	return default_theme
+end
+
+function ui.set_default_theme(theme)
```

Comment:
Error check here to ensure theme is of type ui.Style {}

Reply:
Style objects are immutable, so no need to copy.

There's a lot of places I could do type assertions, but I currently don't. Arguably that's not the best, and I could add it everywhere.

Reply:
Type assertions have been added to all public functions.